src/build_histogram.jl

```diff
@@ -127,23 +146,50 @@ edges, counts  = build_histogram(r, 256, minval = 0, maxval = 1)
 # References
 [1] E. Herrholz, "Parsimonious Histograms," Ph.D. dissertation, Inst. of Math. and Comp. Sci., University of Greifswald, Greifswald, Germany, 2011.
 """
-function build_histogram(img::AbstractArray, nbins::Integer = 256;
+function build_histogram(img::GenericGrayImage, nbins::Integer = 256;
                          minval::Union{Real,AbstractGray}=minfinite(img),
                          maxval::Union{Real,AbstractGray}=maxfinite(img))
     edges = partition_interval(nbins, minval, maxval)
     build_histogram(img, edges)
 end
 
-function build_histogram(img::AbstractArray{T}, edges::AbstractRange) where {T<:Color3}
-    build_histogram(Gray.(img),edges)
+function build_histogram(img::AbstractArray{C}, nbins::Integer = 256; kwargs...) where C<:Color
+    build_histogram(mappedarray(Gray{eltype(C)}, img), nbins; kwargs...)
+end
+
+# Performance optimizations for build_histogram(img, nbins)
+
+function build_histogram(img::AbstractArray{T}) where T<:Union{N0f8, AbstractGray{N0f8}}
+    edges = range(N0f8(0), step=eps(N0f8), length=256)
+    counts = fill(0, 0:256)
+    @simd for v in img
+        @inbounds counts[reinterpret(gray(v))+1] += 1
+    end
+    return edges, counts
+end
+
+build_histogram(img::AbstractArray{C}) where C<:Color{N0f8} =
+    build_histogram(mappedarray(Gray{N0f8}, img))
+
+function build_histogram(img::AbstractArray{T}, nbins::Integer;
+                         minval::Union{Real,AbstractGray}=minfinite(img),
+                         maxval::Union{Real,AbstractGray}=maxfinite(img))  where T<:Union{N0f8, AbstractGray{N0f8}}
+    edgesraw, countsraw = build_histogram(img)
+    return rebin(edgesraw, countsraw, nbins, minval, maxval)
+end
+
+# build_histogram(img, edges)
```

Comment:
Did you intend to leave this comment here?

Reply:
Yeah, kind of a "section header" (the ones above use `nbins`, the ones below `edges`). But I can get rid of it if you don't like it.

Reply:
It is fine. I assumed you meant it as a section header, but just wanted to confirm.